src/views/AddAccountSelection.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+    <div class="container pad-bottom">
+        <SmallPage>
+            <IdenticonSelector  v-if="!showLoader" :accounts="accounts" @identicon-selected="identiconSelected">
+                <PageHeader backArrow @back="back">Choose a new Account</PageHeader>
+            </IdenticonSelector>
+
+            <Loader v-else :title="title" :state="state" :lightBlue="true">
+                <template slot="success">
+                    <CheckmarkIcon/>
+                    <h1 class="title nq-h1">New address added<br>to your account.</h1>
+                </template>
+            </Loader>
+        </SmallPage>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import { SmallPage, PageHeader, CheckmarkIcon } from '@nimiq/vue-components';
+import IdenticonSelector from '../components/IdenticonSelector.vue';
+import { AccountInfo } from '../lib/AccountInfo';
+import { State } from 'vuex-class';
+import { WalletStore } from '../lib/WalletStore';
+import { DerivedAddress } from '@nimiq/keyguard-client';
+import { ParsedSimpleRequest } from '../lib/RequestTypes';
+import { Address } from '../lib/PublicRequestTypes';
+import Loader from '../components/Loader.vue';
+import { Static } from '../lib/StaticStore';
+
+@Component({components: {Loader, SmallPage, PageHeader, IdenticonSelector, CheckmarkIcon}})
+export default class AddAccountSelection extends Vue {
+    @Static private request!: ParsedSimpleRequest;
+    @State private keyguardResult!: DerivedAddress[];
+
+    private showLoader: boolean = false;
+    private state: Loader.State = Loader.State.LOADING;
+    private title: string = 'Storing your address';
+
+    private get accounts(): AccountInfo[] {
+        return this.keyguardResult.map((address) => new AccountInfo(
+            address.keyPath,
+            '',
+            new Nimiq.Address(address.address),
+        ));
+    }
+
+    private back() {
+        window.history.back();
+    }
+
+    private async identiconSelected(selectedAccount: AccountInfo) {
+        this.showLoader = true;
+
+        const wallet = (await WalletStore.Instance.get(this.request.walletId))!;
+
+        if (!selectedAccount.label) {
+            selectedAccount.label = selectedAccount.defaultLabel;
+        }
+
+        wallet.accounts.set(selectedAccount.userFriendlyAddress, selectedAccount);
+
+        await WalletStore.Instance.put(wallet);
+        // Artificially delay, to display loading status
+        await new Promise((res) => setTimeout(res, 1000));
```

Comment:
Why display the loading status at all?
Display the success screen immediately after the wallet was stored and don't even go into the loading state as storing the wallet is very quick.

Reply:
Except when it isn't and the database is blocked for whatever reason.

Reply:
I think it's unlikely that that happens. Not worth letting all users wait 1s.
If you really want handle the case of a long pending database transaction, check after a short delay (maybe 100ms) whether the wallet was stored and only if not display the loading state.